### PR TITLE
TST: Split and simplify test_value_counts_unique_nunique

### LINKED ISSUE #32205: Simplify test_value_counts_unique_nunique_null in pandas/tests/base/test_ops.py
As stated by @jreback in [his comment](https://github.com/pandas-dev/pandas/pull/32046#discussion_r383021295) on #32046, there are some weird constructs, duplications and unreachable code in this test.

For more info see the mentioned PR and the unresolved comments in the bottom.

The other tests in this file have a similar problem, but should probably be a separate PR.

### LINKED ISSUE #32220: CI: value_counts tests failing sometimes
It seems like we're getting some failures from `test_value_counts_unique_nunique` in `pandas/tests/base/test_ops.py` (weird thing is that it only fails some of the time; maybe because of randomized test data?). Looks like these tests were updated in https://github.com/pandas-dev/pandas/pull/32046; @SaturnFromTitan any ideas what might be happening there?

Sample failed pipeline: https://github.com/pandas-dev/pandas/runs/464853016

### PR DESCRIPTION
**Pull request by SaturnFromTitan**
closes #32205, closes #32220

- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by SaturnFromTitan** [comment]
The failing CI is unrelated to these changes

File: pandas/tests/base/test_ops.py
Comment (jreback) [inline pandas/tests/base/test_ops.py]: can you split the test right below as well (alternatively maybe able to combine with the above as they are duplicative, but no big deal)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/base/test_ops.py
Comment (SaturnFromTitan) [reply pandas/tests/base/test_ops.py]: @jreback Done - please have another look. I think the `test_x_null` cases are sufficiently different to not merge them with the non-null ones.

**Comment by SaturnFromTitan** [comment]
@jreback merged master to resolve conflicts

File: pandas/conftest.py
Comment (jreback) [inline pandas/conftest.py]: why remove this?

File: pandas/tests/base/test_ops.py
Comment (jreback) [inline pandas/tests/base/test_ops.py]: why do you need dict.fromkeys?

File: pandas/tests/base/test_ops.py
Comment (jreback) [inline pandas/tests/base/test_ops.py]: can you use nulls_fixture

File: pandas/tests/base/test_ops.py
Comment (jreback) [inline pandas/tests/base/test_ops.py]: same

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/base/test_ops.py
Comment (SaturnFromTitan) [reply pandas/tests/base/test_ops.py]: It seemed like the easiest way to get the unique values of obj while preserving the order. I'm open to alternatives though

File: pandas/tests/base/test_ops.py
Comment (SaturnFromTitan) [reply pandas/tests/base/test_ops.py]: I tried using `nulls_fixture` and `unique_nulls_fixture`. Nearly all tests break for `pd.NaT` and `pd.NA` though...

File: pandas/conftest.py
Comment (SaturnFromTitan) [reply pandas/conftest.py]: No good reason, I'll undo it

File: pandas/tests/base/test_ops.py
Comment (jreback) [reply pandas/tests/base/test_ops.py]: ok, yeah we need to test these, can you create an issue. we will want to add these even if they need xfailing for now as there is no testing on them.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @SaturnFromTitan 

one other followup on this. I think we should split out the tests you just changed (e.g. unique, value_counts) into a test_unique.py module; happy to do this generally as well (though should be separate from a refactorting PR).

File: pandas/tests/base/test_ops.py
Comment (SaturnFromTitan) [reply pandas/tests/base/test_ops.py]: xref: #32437

**Comment by SaturnFromTitan** [comment]
Will do. I'll wait until #32311 is merged though as I think it should go into the test_uniques.py as well

**Pull request merged**
